### PR TITLE
Fix BuildPass>1 artifacts and asset manifest content

### LINKED ISSUE #4778: win-x64 BuildPass=2 vertical's asset manifest contains assets from previously built verticals
Two issues:

1. The vertical manifest that gets produced from the win-x64 BuildPass2 vertical contains all the assets that got produced in the dependent verticals: win-x64, win-x86 and win-arm64. Only two packages should show up there (VS.Redist.*) today. Example: https://gist.github.com/ViktorHofer/484bec9417a95e701c190ba25a5ef4c5
2. Not just the vertical merged manifest is wrong, also the repo specific manifest output is wrong. Example:
```
2024-12-02T12:22:32.1944797Z   New artifact(s) after building sdk:
2024-12-02T12:22:32.1945222Z     -> Aspire.Tools.Service/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1946994Z     -> dotnet-format/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1948783Z     -> FSharp.Compiler.Service/43.9.200-preview.24578.1
2024-12-02T12:22:32.1950361Z     -> FSharp.Core/9.0.200-beta.24578.1
2024-12-02T12:22:32.1952259Z     -> Microsoft.DotNet.ApiCompat.Task/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1954329Z     -> Microsoft.DotNet.ApiCompat.Tool/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1956042Z     -> Microsoft.DotNet.Cli/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1957564Z     -> Microsoft.DotNet.Cli.Sln.Internal/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1959537Z     -> Microsoft.DotNet.Cli.Utils/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1960967Z     -> Microsoft.DotNet.Common.ItemTemplates/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1962898Z     -> Microsoft.DotNet.Common.ProjectTemplates.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1965261Z     -> Microsoft.DotNet.Configurer/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1967774Z     -> Microsoft.DotNet.GenAPI.Task/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1970221Z     -> Microsoft.DotNet.GenAPI.Tool/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1972109Z     -> Microsoft.DotNet.InternalAbstractions/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1974141Z     -> Microsoft.NET.Build.Containers/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1976233Z     -> Microsoft.NET.Build.Extensions/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1977937Z     -> Microsoft.NET.Sdk/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1980265Z     -> Microsoft.NET.Sdk.BlazorWebAssembly/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1982106Z     -> Microsoft.Net.Sdk.Compilers.Toolset/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1983912Z     -> Microsoft.NET.Sdk.Publish/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1986811Z     -> Microsoft.NET.Sdk.Razor/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1989281Z     -> Microsoft.NET.Sdk.StaticWebAssets/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1992436Z     -> Microsoft.NET.Sdk.Web/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1994969Z     -> Microsoft.NET.Sdk.Web.ProjectSystem/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1997210Z     -> Microsoft.NET.Sdk.WebAssembly/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.1999950Z     -> Microsoft.NET.Sdk.Worker/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2002766Z     -> Microsoft.NET.Sdk.WorkloadManifestReader/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2005217Z     -> Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2007699Z     -> Microsoft.TemplateEngine.Cli/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2049272Z     -> VS.Redist.Common.NetCore.SdkPlaceholder.arm64.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2049685Z     -> VS.Redist.Common.NetCore.Templates.arm64.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2050149Z     -> VS.Redist.Common.NetCore.Toolset.arm64.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2050623Z     -> VS.Redist.Common.NetCore.SdkPlaceholder.x64.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2050980Z     -> VS.Redist.Common.NetCore.Templates.x64.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2051567Z     -> VS.Redist.Common.NetCore.Toolset.x64.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2051877Z     -> VS.Redist.Common.Net.Core.SDK.MSBuildExtensions/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2052193Z     -> VS.Redist.Common.Net.Core.SDK.VSTemplateLocator/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2052522Z     -> VS.Redist.Common.NetCore.SdkPlaceholder.x86.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2052842Z     -> VS.Redist.Common.NetCore.Templates.x86.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2053145Z     -> VS.Redist.Common.NetCore.Toolset.x86.10.0/10.0.100-alpha.1.24602.1
2024-12-02T12:22:32.2053467Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-toolset-internal-10.0.100-alpha.1.24602.1.zip
2024-12-02T12:22:32.2053816Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-toolset-langpack-10.0.100-alpha.1.24602.1.zip
2024-12-02T12:22:32.2054144Z     -> assets/symbols/Aspire.Tools.Service.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2054470Z     -> assets/symbols/dotnet-format.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2054794Z     -> assets/symbols/Microsoft.DotNet.ApiCompat.Task.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2055132Z     -> assets/symbols/Microsoft.DotNet.ApiCompat.Tool.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2055441Z     -> assets/symbols/Microsoft.DotNet.Cli.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2055768Z     -> assets/symbols/Microsoft.DotNet.Cli.Sln.Internal.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2056106Z     -> assets/symbols/Microsoft.DotNet.Cli.Utils.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2056461Z     -> assets/symbols/Microsoft.DotNet.Common.ItemTemplates.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2065116Z     -> assets/symbols/Microsoft.DotNet.Common.ProjectTemplates.10.0.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2075552Z     -> assets/symbols/Microsoft.DotNet.Configurer.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2083061Z     -> assets/symbols/Microsoft.DotNet.GenAPI.Task.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2087972Z     -> assets/symbols/Microsoft.DotNet.GenAPI.Tool.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2092783Z     -> assets/symbols/Microsoft.DotNet.InternalAbstractions.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2096799Z     -> assets/symbols/Microsoft.NET.Build.Containers.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2100744Z     -> assets/symbols/Microsoft.NET.Build.Extensions.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2104927Z     -> assets/symbols/Microsoft.NET.Sdk.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2108871Z     -> assets/symbols/Microsoft.NET.Sdk.BlazorWebAssembly.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2112700Z     -> assets/symbols/Microsoft.Net.Sdk.Compilers.Toolset.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2117147Z     -> assets/symbols/Microsoft.NET.Sdk.Publish.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2120923Z     -> assets/symbols/Microsoft.NET.Sdk.Razor.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2126117Z     -> assets/symbols/Microsoft.NET.Sdk.StaticWebAssets.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2128710Z     -> assets/symbols/Microsoft.NET.Sdk.Web.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2131362Z     -> assets/symbols/Microsoft.NET.Sdk.Web.ProjectSystem.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2134119Z     -> assets/symbols/Microsoft.NET.Sdk.WebAssembly.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2136633Z     -> assets/symbols/Microsoft.NET.Sdk.Worker.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2140143Z     -> assets/symbols/Microsoft.NET.Sdk.WorkloadManifestReader.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2143038Z     -> assets/symbols/Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2146949Z     -> assets/symbols/Microsoft.TemplateEngine.Cli.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2150677Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-100templates-10.0.100-alpha.1.24602.1-win-arm64.msi
2024-12-02T12:22:32.2154433Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-100templates-10.0.100-alpha.1.24602.1-win-arm64.msi.wixpack.zip
2024-12-02T12:22:32.2158184Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-10.0.100-alpha.1.24602.1-win-arm64.zip
2024-12-02T12:22:32.2162122Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-10.0.100-alpha.1.24602.1-win-arm64.zip.sha512
2024-12-02T12:22:32.2165458Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-internal-10.0.100-alpha.1.24602.1-win-arm64.msi
2024-12-02T12:22:32.2169012Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-internal-10.0.100-alpha.1.24602.1-win-arm64.msi.wixpack.zip
2024-12-02T12:22:32.2171680Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdkplaceholder-10.0.100-alpha.1.24602.1-win-arm64.msi
2024-12-02T12:22:32.2176286Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdkplaceholder-10.0.100-alpha.1.24602.1-win-arm64.msi.wixpack.zip
2024-12-02T12:22:32.2179752Z     -> Sdk/10.0.100-alpha.1.24602.1/productCommit-win-arm64.json
2024-12-02T12:22:32.2183818Z     -> Sdk/10.0.100-alpha.1.24602.1/productCommit-win-arm64.txt
2024-12-02T12:22:32.2186481Z     -> Sdk/10.0.100-alpha.1.24602.1/win_arm64_Release_version_badge.svg
2024-12-02T12:22:32.2189083Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-100templates-10.0.100-alpha.1.24602.1-win-x64.msi
2024-12-02T12:22:32.2191921Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-100templates-10.0.100-alpha.1.24602.1-win-x64.msi.wixpack.zip
2024-12-02T12:22:32.2194320Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-10.0.100-alpha.1.24602.1-win-x64.zip
2024-12-02T12:22:32.2196892Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-10.0.100-alpha.1.24602.1-win-x64.zip.sha512
2024-12-02T12:22:32.2200010Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-internal-10.0.100-alpha.1.24602.1-win-x64.msi
2024-12-02T12:22:32.2203368Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-internal-10.0.100-alpha.1.24602.1-win-x64.msi.wixpack.zip
2024-12-02T12:22:32.2206441Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdkplaceholder-10.0.100-alpha.1.24602.1-win-x64.msi
2024-12-02T12:22:32.2210838Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdkplaceholder-10.0.100-alpha.1.24602.1-win-x64.msi.wixpack.zip
2024-12-02T12:22:32.2213158Z     -> Sdk/10.0.100-alpha.1.24602.1/productCommit-win-x64.json
2024-12-02T12:22:32.2216261Z     -> Sdk/10.0.100-alpha.1.24602.1/productCommit-win-x64.txt
2024-12-02T12:22:32.2219714Z     -> Sdk/10.0.100-alpha.1.24602.1/productVersion.txt
2024-12-02T12:22:32.2222689Z     -> Sdk/10.0.100-alpha.1.24602.1/sdk-productVersion.txt
2024-12-02T12:22:32.2225452Z     -> Sdk/10.0.100-alpha.1.24602.1/win_x64_Release_version_badge.svg
2024-12-02T12:22:32.2228327Z     -> assets/symbols/VS.Redist.Common.Net.Core.SDK.MSBuildExtensions.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2230394Z     -> assets/symbols/VS.Redist.Common.Net.Core.SDK.VSTemplateLocator.10.0.100-alpha.1.24602.1.symbols.nupkg
2024-12-02T12:22:32.2232332Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-100templates-10.0.100-alpha.1.24602.1-win-x86.msi
2024-12-02T12:22:32.2235293Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-100templates-10.0.100-alpha.1.24602.1-win-x86.msi.wixpack.zip
2024-12-02T12:22:32.2237971Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-10.0.100-alpha.1.24602.1-win-x86.zip
2024-12-02T12:22:32.2240758Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-10.0.100-alpha.1.24602.1-win-x86.zip.sha512
2024-12-02T12:22:32.2243144Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-internal-10.0.100-alpha.1.24602.1-win-x86.msi
2024-12-02T12:22:32.2245845Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdk-internal-10.0.100-alpha.1.24602.1-win-x86.msi.wixpack.zip
2024-12-02T12:22:32.2248637Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdkplaceholder-10.0.100-alpha.1.24602.1-win-x86.msi
2024-12-02T12:22:32.2250553Z     -> Sdk/10.0.100-alpha.1.24602.1/dotnet-sdkplaceholder-10.0.100-alpha.1.24602.1-win-x86.msi.wixpack.zip
2024-12-02T12:22:32.2259916Z     -> Sdk/10.0.100-alpha.1.24602.1/productCommit-win-x86.json
2024-12-02T12:22:32.2260261Z     -> Sdk/10.0.100-alpha.1.24602.1/productCommit-win-x86.txt
2024-12-02T12:22:32.2260565Z     -> Sdk/10.0.100-alpha.1.24602.1/win_x86_Release_version_badge.svg
```

Whatever we do here, we should preserve the following principles:
1. A BuildPass=2 should just be a continuation of the previous run but with more input assets. But we need some way to different previously built assets from fresh built assets.
2. Only the assets that got produced in that vertical should get uploaded.

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/source-build/issues/4778

There are three problems with the current implementation:

1. The asset manifest that gets produced in a BuildPass>2 build contains all the assets from the previous dependent vertical manifests. This is undesirable in general but also causes an issue when we merge all manifests together because there are duplicates that can't be resolved.

2. Artifacts from the dependent verticals get uploaded to the "<JobName>_Artifacts" pipeline artifact in CI in addition to the new artifacts produced which is a waste of resources. It also makes it harder to differentiate what actually got produced in the BuildPass>1 leg.

3. The log that lists all the artifacts that got produced by the repo build also outputs the artifacts from the dependent verticals because their manifests were considered as well.

4. Repo asset manifests from dependent verticals have the same name and therefore got ignored. Only the first one gets copied, all the other repo asset manifest files are ignored.

---

Isolate the asset manifests that get produced by the build and then later read to generated the merged (vertical) manifest, from the downloaded dependent vertical manifests. Upload the vertical manifest only, instead of all the inner repo manifests. Name the manifest based on the job name to avoid conflicts.

Don't upload everything under artifacts/assets and artifacts/packages. Instead read from the current vertical merged manifest to identify the assets to upload. Leverage an msbuild target to bin place the artifacts into the staging directory to upload. As part of that, also introduce a YML configuration property that defaults to Release and pass that into the build to be able to find the path to the vertical manifest without hardcoding a certain configuration.

---

Here's a build from this PR that demonstrates that the changes work as expected: https://dnceng-public.visualstudio.com/public/_build/results?buildId=888307&view=logs&s=56cc5bc2-3154-5d19-920f-2db03d3b8537&j=287a7939-eab3-5c36-8f0e-00afcd687924

### CODE REVIEW

**Comment by ViktorHofer** [comment]
This one is ready. @mmitche PTAL

File: eng/pipelines/templates/jobs/vmr-build.yml [hunk 1]
```diff
@@ -158,6 +162,11 @@ jobs:
     - name: sourcesPath
       value: $(vmrPath)
 
+  # Must be a path under $(sourcesPath). Inside the docker container, we mount $(sourcesPath) to /vmr
+  # and can't write outside of that folder.
+  - name: artifactsStagingDir
+    value: $(sourcesPath)/artifacts/staging
```
Comment (mmitche) [inline eng/pipelines/templates/jobs/vmr-build.yml:168]: If we weren't running our own custom docker process, could we default this to AzDO's Artifacts.StagingDirectory?

Comment (ViktorHofer) [reply eng/pipelines/templates/jobs/vmr-build.yml:168]: Yes

File: src/SourceBuild/content/build.proj [hunk 2]
```diff
@@ -26,7 +26,8 @@
     <ProjectReference Include="$(RepositoryEngineeringDir)finish-source-only.proj" Condition="'$(DotNetBuildSourceOnly)' == 'true'" />
   </ItemGroup>
 
-  <Target Name="PrintInfo">
+  <!-- Don't print this during restore. -->
+  <Target Name="PrintInfo" Condition="'$(MSBuildRestoreSessionId)' == ''">
```
Comment (mmitche) [inline src/SourceBuild/content/build.proj:30]: Weird, I didn't know this was possible. Does this still print info more than once?

Comment (mmitche) [reply src/SourceBuild/content/build.proj:30]: Why not just have it as a BeforeTargets="Build" (and remove the InitialTargets)

File: src/SourceBuild/content/eng/finish-source-only.proj [hunk 3]
```diff
@@ -16,14 +16,14 @@
 
   <!-- After building, generate a prebuilt usage report. -->
   <Target Name="ReportPrebuiltUsage"
-          AfterTargets="Build"
+          BeforeTargets="Build"
```
Comment (mmitche) [inline src/SourceBuild/content/eng/finish-source-only.proj:19]: Were these running in the wrong place, or is this just clean up?

File: src/SourceBuild/content/build.proj (hunk 2 above)
Comment (ViktorHofer) [reply src/SourceBuild/content/build.proj]: I remember there was a reason back when we added the target but I don't see any reason to not do this with the current code. Thanks, updated.

File: src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/GetKnownArtifactsFromAssetManifests.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+        /// <summary>
+        /// The list of known blobs.
+        /// </summary>
+        [Output]
+        public ITaskItem[]? KnownBlobs { get; set; }
+
+        public override bool Execute()
+        {
+            XDocument[] xDocuments = AssetManifests
```
Comment (mmitche) [inline src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/GetKnownArtifactsFromAssetManifests.cs:58]: There are at least 3-5 implementations of reading/writing the asset manifest content in the VMR and arcade (and maybe even Maestro, sigh). https://github.com/dotnet/arcade/issues/15317

File: src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/GetKnownArtifactsFromAssetManifests.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                .Select(manifest => XDocument.Load(manifest.ItemSpec))
+                .ToArray();
+
+            KnownPackages = xDocuments
+                .SelectMany(doc => doc.Root!.Descendants(PackageElementName))
+                .Where(ShouldIncludeElement)
+                .Distinct()
+                .Select(package => new TaskItem(package.Attribute(IdAttributeName)!.Value, new Dictionary<string, string>
+                {
+                    { PackageVersionAttributeName, package.Attribute(PackageVersionAttributeName)!.Value },
```
Comment (mmitche) [inline src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/GetKnownArtifactsFromAssetManifests.cs:68]: Are you mainly adding the additional attributes here for completeness?

**Review by mmitche — APPROVED** [review]

File: src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/GetKnownArtifactsFromAssetManifests.cs (hunk 5 above)
Comment (ViktorHofer) [reply src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/GetKnownArtifactsFromAssetManifests.cs:68]: Correct. We only need the NonShipping, RepoOrigin and Version ones. The `DotNetReleaseShipping` is just here for completeness. The other changes in this file allow filtering artiafcts based on `RepoOrigin` and enabling `#nullable enable` to check for nullrefs.

**Pull request merged**